File: cpp/task2.cpp
#include <iostream>
#include <vector>
#include <tuple>
#include <algorithm>

int main() {
    int N;
    std::cin >> N;
    std::vector<std::tuple<int,int, int>> meet;
    for (int i = 0; i < N; ++i) {
        int start, end;
        std::cin >> start >> end;
        if (start >= end){
            continue;
        }
        meet.push_back({end, start, i + 1});
    }
    if (meet.empty()) {
        std::cout << "0" << std::endl;
        return 0;
    }
    std::sort(meet.begin(), meet.end());
    std::vector<int> result;
    int lastEnd = 0;
    for (const auto& m : meet) {
        int end = std::get<0>(m);
        int start = std::get<1>(m);
        int index = std::get<2>(m);
        if (start >= lastEnd) {
            result.push_back(index);
            lastEnd = end;
        }
    }
    std::sort(result.begin(), result.end());
    std::cout << result.size() << std::endl;
    for (size_t i = 0; i < result.size(); ++i) {
        if (i > 0) std::cout << " ";
        std::cout << result[i];
    }
    std::cout << std::endl;
    return 0;
}

File: rust/src/task_2.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();    let n: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
    let mut meet: Vec<(i32, i32, i32)> = (0..n)
    .filter_map(|i| {
        let line = lines.next().unwrap().unwrap();
        let parts: Vec<i32> = line.split_whitespace().map(|x|x.parse().unwrap()).collect();
        let (start, end) = (parts[0], parts[1]);
        if start < end {
            Some((end, start, (i + 1) as i32))
        } else {
            None
        }
    })
    .collect();
    if meet.is_empty() {
        println!("0");
        return;
    }
    meet.sort();
    let mut result = Vec::new();
    let mut last_end = 0;
    for &(end, start, index) in &meet {
        if start >= last_end {
            result.push(index);
            last_end = end;
        }
    }
    result.sort();
    println!("{}", result.len());
    for (i, &idx) in result.iter().enumerate() {
        if i > 0 { print!(" "); 
    }
    print!("{}", idx);}
    println!();
}